### PR TITLE
Importable e2e tests

### PR DESCRIPTION
**Pull request by smarshall-spitzbart**
Allows e2e tests to be imported into external repos, and ran against any arbitrary `app.go` that implements the necessary interfaces

### CODE REVIEW

File: tests/e2e/democracy.go [hunk 1]
```diff
@@ -30,28 +26,41 @@ var consumerFraction, _ = sdk.NewDecFromStr(consumerkeeper.ConsumerRedistributeF
 type ConsumerDemocracyTestSuite struct {
 	suite.Suite
 	coordinator   *ibctesting.Coordinator
-	providerChain *ibctesting.TestChain
```
Comment (smarshall-spitzbart) [inline tests/e2e/democracy.go:33]: Removed since none of the democracy tests use it

File: tests/e2e/instance_test.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	appConsumer "github.com/cosmos/interchain-security/app/consumer"
+	appConsumerDemocracy "github.com/cosmos/interchain-security/app/consumer-democracy"
+	appProvider "github.com/cosmos/interchain-security/app/provider"
+	"github.com/cosmos/interchain-security/tests/e2e"
+	e2etestutil "github.com/cosmos/interchain-security/testutil/e2e"
+	"github.com/cosmos/interchain-security/testutil/simapp"
+	"github.com/stretchr/testify/suite"
+)
+
+// This file can be used as an example e2e testing instance for any provider/consumer applications.
```
Comment (MSalopek) [inline tests/e2e/instance_test.go:16]: This is a really nice overview on how to use this and it communicates the requrements in a clear and concise manner.

Maybe consider adding it to the `tests/e2e` readme.

Comment (smarshall-spitzbart) [reply tests/e2e/instance_test.go:16]: Agreed, I've added instructions to the top level readme, and the `tests/e2e` has a quick mention as well

**Review by jtremback — APPROVED** [review]
LGTM

**Review by mpoke — APPROVED** [review]
LGTM. Just one nit.

File: tests/e2e/README.md [hunk 3]
```diff
@@ -10,4 +10,6 @@ E2e tests are categorized into files as follows:
 - `slashing_test.go` - e2e tests for the _Consumer Initiated Slashing_ sub-protocol
 - `distribution_test.go` - e2e tests for the _Reward Distribution_ sub-protocol
 - `stop_consumer_test.go` - e2e tests for the _Consumer Chain Removal_ sub-protocol
-- `normal_operations_test.go` - e2e tests for _normal operations_ of ICS enabled chains 
\ No newline at end of file
+- `normal_operations_test.go` - e2e tests for _normal operations_ of ICS enabled chains
```
[the remark this answers is not in the document]
Comment (smarshall-spitzbart) [reply tests/e2e/README.md:13]: Nice catch 👍

**Pull request merged**